Source/displayinfo/include/ExtendedDisplayIdentification.h

```diff
@@ -303,31 +703,126 @@ namespace Plugin {
+            colorDepthMap |= static_cast<uint8_t>(ColorDepth());
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
+                    CEA cae(segment.Current());
+                    cae.SupportedColorDepths(colorDepthMap);
+                }
+            }
+        }
+
+        ColorFormatType SupportedColorFormat() const {
+            ColorFormatType result = ColorFormatType::UNDEFINED;
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
+                    CEA cae(segment.Current());
+                    result = cae.SupportedColorFormat();
+                }
+            }
+
+            return result;
+        }
+
+        void SupportedDigitalDisplayTypes(uint8_t& colorFormatMap) const {
+            if((_base[0x14] >> 7) & 0x01){
+                switch((_base[0x18] >> 3) & 0x07) {
+                    case 0x00:
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::RGB);
+                        break;
+                    case 0x01:
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::YCBCR_4_4_4);
+                        break;
+                    case 0x02:
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::YCBCR_4_2_2);
+                        break;
+                    case 0x03:
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::YCBCR_4_2_2);
+                        colorFormatMap |= static_cast<uint8_t>(ColorFormatType::YCBCR_4_4_4);
+                        break;
+                }
+            }
+        }
+
+        void SupportedColorFormats(uint8_t& colorFormatMap) const {
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
+                    CEA cae(segment.Current());
+                    colorFormatMap |= static_cast<uint8_t>(cae.SupportedColorFormat());
+                    cae.SupportedColorFormats(colorFormatMap);
+                }
+            }
+            SupportedDigitalDisplayTypes(colorFormatMap);
+        }
+
+        void SupportedColorSpace(uint16_t& colorSpaceMap) const {
+            if((_base[0x18] & (1 << 2))) {
+                colorSpaceMap |= static_cast<uint16_t>(ColorSpaceType::SRGB);
+            }
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
```

Comment:
Can there be many of them?

Reply:
Frankly speaking have no idea, never met second CEA block. Added break on the first finding anyway